### ISSUE TITLE
Incorrect permitted-contents error: option in datalist

### ISSUE THREAD

**Issue by yonta** [markuplint/markuplint#1146]
- Markuplint version: `3.12.1`
- Parser lang: HTML
- Node.js version: `18.17.1`
- OS: Windows11 with WSL2 Ubuntu 22.04.2 LTS

## Describe the bug

The datalist element which has option element occure error.
But it is valid HTML.

## Code Example

```html
    <label for="ice-cream-choice">Choose a flavor:</label>
    <input list="ice-cream-flavors" id="ice-cream-choice" name="ice-cream-choice" />
    <datalist id="ice-cream-flavors" aria-label="ice cream flavors">
      <option value="Chocolate" aria-label="chocolate"></option>
    </datalist>
```

```json
// .markuplintrc
{
  "extends": ["markuplint:recommended"],
  "rules": {
    "permitted-contents": [
      {
        "tag": "datalist",
        "contents": [
          {
            "zeroOrMore": "option"
          }
        ]
      }
    ]
  }
}
```

## Steps To Reproduce

1. run markuplint

## The current behavior

This occures permitted-contents error.

```console
$ yarn markuplint ./test.html 
<markuplint> error: このコンテキストでは、要素「datalist」に要素「option」を含めることはできません (permitted-contents) 
/[[MYPATH]]/test.html:12:7
  11: ••••<datalist•id="ice-cream-flavors"•aria-label="hoge">
  12: ••••••<option value="Chocolate"></option>
  13: ••••</datalist>
```
## The expected behavior

The datalist element can have option elements.

https://html.spec.whatwg.org/multipage/form-elements.html#the-datalist-element

## Other Question

Can I change the behavior of permitted-contents for regular (not custom) tags (datalist, etc.) by some configurations, or not?

**Comment by YusukeHirao** [comment]
@yonta 

Thank you. I'll check it.

> Can I change the behavior of permitted-contents for regular (not custom) tags (datalist, etc.) by some configurations, or not?

It might not work. I'm pretty sure didn't check it and didn't write a test.

**Comment by YusukeHirao** [comment]
The schema of the `datalist` element.

https://github.com/markuplint/markuplint/blob/f6e0b456f01fba353f385eae5a7f3d560bc03bba/packages/%40markuplint/html-spec/src/spec.datalist.json#L6-L21C4

But, the `choice` pattern had a bug in its computation.
I've pinpointed the issue up to there.
